tests/unit/api/multipartUpload.js

```diff
@@ -2070,6 +2066,175 @@ describe('Multipart Upload API', () => {
             });
         });
     });
+
+    it('should return a retryable error if deletePartsMetadata fails', done => {
+        const partBody = Buffer.from('I am a part\n', 'utf8');
+        let batchDeleteStub;
+
+        async.waterfall([
+            next => bucketPut(authInfo, bucketPutRequest, log, next),
+            (corsHeaders, next) => initiateMultipartUpload(authInfo,
+                initiateRequest, log, next),
+            (result, corsHeaders, next) => parseString(result, next),
+        ],
+        (err, json) => {
+            assert.ifError(err);
+            const testUploadId = json.InitiateMultipartUploadResult.UploadId[0];
+            const partHash = crypto.createHash('md5').update(partBody).digest('hex');
+            const partRequest = new DummyRequest({
+                bucketName,
+                namespace,
+                objectKey,
+                headers: { host: `${bucketName}.s3.amazonaws.com` },
+                url: `/${objectKey}?partNumber=1&uploadId=${testUploadId}`,
+                query: {
+                    partNumber: '1',
+                    uploadId: testUploadId,
+                },
+                partHash,
+            }, partBody);
+
+            objectPutPart(authInfo, partRequest, undefined, log, () => {
+                // Mock batchDeleteObjectMetadata to fail with non-retryable error
+                const services = require('../../../lib/services');
+                batchDeleteStub = sinon.stub(services, 'batchDeleteObjectMetadata')
+                    .callsFake((mpuBucketName, keysToDelete, log, cb) =>
+                        // Simulate a non-retryable error that should be converted to retryable
+                         cb(errors.NoSuchKey)
```

Comment:
```suggestion
                const services = require('../../../lib/services');
                batchDeleteStub = sinon.stub(services, 'batchDeleteObjectMetadata')
                    .callsFake((mpuBucketName, keysToDelete, log, cb) =>
                         cb(errors.NoSuchKey)
```

Your code is clear enough